tensorboard/webapp/widgets/filter_input/filter_input_component.ts

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, Input, ViewChild} from '@angular/core';
+import {MatAutocompleteTrigger} from '@angular/material/autocomplete';
+
+/**
+ * A text input field intended for filtering items.
+ */
+@Component({
+  selector: 'tb-filter-input',
+  template: `
+    <input
```

Comment:
I noticed that the two applications both have the `<mat-icon svgIcon="search_24px">`. Would it maybe make sense to include the icon since it is literally for filters?

It would be close to this (on older safari): https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search
![image](https://user-images.githubusercontent.com/2547313/118226247-74c3ef00-b43b-11eb-93df-07f7787d9ede.png)